rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -25,3 +47,95 @@ def __init__(
             pool_address=POOL_ADDRESS,
             eth_gateways=ETH_GATEWAYS,
         )
+
+    def _decode_staking_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode aave v2 staking unstaking events"""
+        if context.tx_log.topics[0] not in {STAKED_AAVE, REDEEM_AAVE}:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+        to_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if not self.base.any_tracked([from_address, to_address]):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        if context.tx_log.topics[0] == STAKED_AAVE:
```

Comment:
you are repeating the same check as above here. Not needed. In the beginning do something like:

```python
if STAKED_AAVE:
    method = self._decode_stake
elif REDEEM_AAVE:
    method = self._decode_unstake
```

And then here just call the method. But please with kwargs.